spec/p2p/reactor-survey.md

```diff
@@ -0,0 +1,247 @@
+> The above property derives from the requirement that new validators can join consensus. In order to do so, they must first be connected.
+
+> TODO: what do we need to say about new nodes joining and existing nodes resyncing. -> requirement to establish new connections needa to be captured.
+
+Since the systems we are building are decentralized and distributed, the global requirements can only be ensured by local actions of the distributed nodes. For instance, openness has been ensured by distinguishing inbound and outbound connections, and making sure that there are always nodes with open inbound connections.
+
+> The consensus properties and the consensus algorithm are clearly specified in the arXiv paper. In the this section we could thus start our discussions from the consensus gossip property that provides a clear abstraction. As this level of detail of specification is missing for the other reactors, we need to survey them a bit in the following in more detail
+
+## Mempool
+
+Considering the whole CometBFT system, intuitively, there should be end-to-end 
+requirements regarding transaction processing, e.g., "every transaction submitted
+should be eventually put into a block" or "every transaction should be put in
+at most one block". Note that these are just simple examples that are neither ensured
+by current implementation nor are actually required by the system (e.g., typically we 
+do not care about "all" transaction but rather "all valid"). 
+Which of such requirements are actually ensured by the 
+system depends on the guarantees by the concerned parts of the system.
+
+The mempool is a distributed pool of "pending" transactions.
+A pending transaction is a "valid" transaction that has been submitted by a
+client of the blockchain but has not yet been committed to the blockchain.
+The mempool is thus fed with client transactions,
+that a priori can be submitted to any node in the network.
+And it is consumed by the consensus protocol, more specifically by validator nodes,
+which retrieve from the mempool transactions to be included in proposed blocks.
+
+> Note that in this document, we want to capture (an overapproximation of) the requirements the mempool puts on p2p. The notions of "pending" and "valid" seem mostly to be conditions that the mempool can use to drop transactions (and exempt them from beeing spread over the network). Thus for the purpose of this document we will largely ignore them (they should eventually be addressed [here](https://github.com/cometbft/cometbft/issues/223)). 
+
+As 
+- full nodes add transactions to the mempool following client requests, and
+- validators read transactions from the mempool to compose proposed blocks,
+
+the mentioned end-to-end requirements on transactions translate into requirements 
+on the mempool, e.g., every transaction that is submitted to a full node should 
+eventually be present at the mempool of a validator/proposer. 
```

Comment:
No way to guarantee this property.

Reply:
Thanks for looking at the document and giving feedback. Please note that this is a working document that should describe requirements one may expect from the reactors in order to formalize, in turn, what the reactors expect from the p2p system below. 

By describing stronger properties (than those we actually need) here, and deriving properties for p2p from them, we can eventually specify a p2p system that may do a bit more than is needed and thus suffices for what the reactors need in reality. We had noted that on line 104, which says 

> Note that in this document, we want to capture (an overapproximation of) the requirements the mempool puts on p2p. 

We will try to make this clearer in a revision to avoid future confusion.

Reply:
That's all fine, I think. But it is not the case that "every transaction that is submitted to a full node should eventually be present at the mempool of a validator/proposer".